File: rest-service/src/main/java/com/mdbank/controller/LocalDataController.java
package com.mdbank.controller;

import com.mdbank.model.Position;
import com.mdbank.model.metadata.MeteoParameter;
import com.mdbank.model.metadata.NetCdfParam;
import com.mdbank.service.LocalDataService;
import io.swagger.annotations.ApiOperation;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import java.time.LocalDateTime;
import java.util.Map;

@RestController
@RequestMapping("${api.root}/local-data")
public class LocalDataController {
    @Autowired
    LocalDataService localDataService;

    @RequestMapping(method = RequestMethod.GET, value = "/{latitude}/{longitude}/{parameter}/{year}")
    @ApiOperation("Получить данные для конкретной точки")
    public ResponseEntity<Map<LocalDateTime, Float>> getLocalData(
            @PathVariable double latitude,
            @PathVariable double longitude,
            @PathVariable NetCdfParam parameter,
            @PathVariable int year) {
        Position position = new Position(latitude, longitude);
        return new ResponseEntity<>(localDataService.getBy(position, parameter, year), HttpStatus.OK);
    }

}

File: rest-service/src/main/java/com/mdbank/controller/LocalDataController.kt
package com.mdbank.controller

import com.mdbank.controller.dto.localdata.FormattedLocalData
import com.mdbank.service.LocalDataService
import io.swagger.annotations.ApiOperation
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.transaction.annotation.Transactional
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController
import java.time.Year

@RestController
@RequestMapping("\${api.root.v1}/local-data")
class LocalDataController {
    @Autowired
    private lateinit var localDataService: LocalDataService

    @GetMapping
    @ApiOperation("Поиск данных для конкретной географической точки, года и метео параметра")
    @Transactional(readOnly = true)
    fun getLocalDataByPositionAndYear(@RequestParam latitude: Double,
                                      @RequestParam longitude: Double,
                                      @RequestParam year: Year,
                                      @RequestParam parameter: String): ResponseEntity<FormattedLocalData> {
        val dataByPositionAndYear = localDataService.getFormattedDataByPositionAndYear(latitude, longitude, year, parameter)
        return dataByPositionAndYear?.let { ResponseEntity.ok(dataByPositionAndYear) }
                ?: ResponseEntity(HttpStatus.NOT_FOUND)
    }
}